runners/flink/src/main/java/org/apache/beam/runners/flink/metrics/FlinkMetricContainer.java

```diff
@@ -77,12 +82,72 @@ public FlinkMetricContainer(RuntimeContext runtimeContext) {
     this.metricsAccumulator = (MetricsAccumulator) metricsAccumulator;
   }
 
-  MetricsContainer getMetricsContainer(String stepName) {
+  public MetricsContainer getMetricsContainer(String stepName) {
     return metricsAccumulator != null
         ? metricsAccumulator.getLocalValue().getContainer(stepName)
         : null;
   }
 
+  /**
+   * Parse a {@link MetricName} from a {@link
+   * org.apache.beam.model.fnexecution.v1.BeamFnApi.MonitoringInfoUrns.Enum}
+   *
+   * <p>Should be consistent with {@code parse_namespace_and_name} in monitoring_infos.py
+   *
+   * <p>TODO: not flink-specific; where should it live?
```

Comment:
Flagging this as well, in case anyone has ideas.

I would normally view this as a`MetricName` constructor, but `MetricName` is in java-core, which doesn't depend on fn-api (and shouldn't be made to without good reason, I assume).

Reply:
I believe that should be changed in `BeamFnApi.MonitoringInfo`.  Ok for now as a workaround.